Refactored and cleaned up code

mzuhl opened the pull request.

mzuhl committed:
Refactored and cleaned up code

mholck commented on BDF-Chainlit-Chatbot/PDC_tools.py:
```diff
@@ -0,0 +1,582 @@
+      response.raise_for_status()
+        
+
+# This is a helper function
```
Should add some description of what this helper function is for in the comment

mzuhl replied:
Updated

mholck commented on BDF-Chainlit-Chatbot/PDC_tools.py:
```diff
@@ -0,0 +1,582 @@
+
+# This is a helper function
+def getDiseaseInformation(mdata:object)->str:
+    '''
```
some comments are before the function using # and others imbedded using ''', should probably come up with a style guide and be consistent.

mholck commented on BDF-Chainlit-Chatbot/PDC_tools.py:
```diff
@@ -0,0 +1,582 @@
+            return {}
+
+            
+post_url = "https://proteomic.datacommons.cancer.gov/graphql"
```
Should this be an environment variable or parameter rather than hard coded?

mzuhl replied:
Updated

mholck commented on BDF-Chainlit-Chatbot/authentication.py:
```diff
@@ -0,0 +1,46 @@
+
+# Configure logging to output to stdout
+logging.basicConfig(
+    level=logging.INFO,  # Log level
```
Should this be an environment variable or parameter rather than hard coded?

mzuhl replied:
Removed logging for now as it was not implemented for the whole app

mholck commented on BDF-Chainlit-Chatbot/chainlit.md:
```diff
@@ -1,14 +1,23 @@
+### Scope
 
-## Useful Links 🔗
+Enables plain-language queries of biomedical data resources – such as CRDC, ProteomeXchange and Metabolomics Workbench. Provides context-aware insights rather than generic responses. Focuses on resource-specific capabilities instead of duplicating broad search functions.	
```
Given that the code only included GDC and PDC is this statement accurate or promising too much

mzuhl replied:
Updated

luke-mcconnell commented on BDF-Chainlit-Chatbot/.env:
```diff
@@ -1,12 +1,31 @@
+# AWS_ACCESS_KEY=""
+# AWS_SECRET_KEY=""
+AWS_REGION="us-east-1"
+PUBLICATIONS_KB_ID="QZB0NZCXSB"
```
```suggestion
PUBLICATIONS_KB_ID=""
```

luke-mcconnell replied:
Public won't have access to our AWS KB

mzuhl replied:
Updated

luke-mcconnell commented on BDF-Chainlit-Chatbot/.env:
```diff
@@ -1,12 +1,31 @@
 
+# Change default model as needed
+DEFAULT_MODEL="us.anthropic.claude-3-5-sonnet-20241022-v2:0"
+DATA_LAYER_TABLE="chainlit_table"
```
```suggestion
DATA_LAYER_TABLE=""
```

luke-mcconnell replied:
Public won't have access to our AWS Dynamodb table

mzuhl replied:
Updated

luke-mcconnell commented on BDF-Chainlit-Chatbot/.env:
```diff
@@ -1,12 +1,31 @@
+# Set your AWS access and secret keys
+# AWS_ACCESS_KEY=""
+# AWS_SECRET_KEY=""
+AWS_REGION="us-east-1"
```
Should this be removed too? Users should probably specify their own region.

mzuhl replied:
Updated

luke-mcconnell commented on BDF-Chainlit-Chatbot/chatbot-server.py:
```diff
@@ -410,6 +365,7 @@ async def rag_query(
         )
         response = query_engine.query(ev.query)
         retrieved_results = response.response
+        breakpoint()
```
```suggestion
```

mzuhl replied:
Removed

luke-mcconnell commented on BDF-Chainlit-Chatbot/tools.py:
```diff
@@ -1,595 +1,12 @@
-
-from typing import Any, Awaitable, Optional, Callable, Type, List, Tuple, Union, cast
-import boto3
+from typing import List
```
```suggestion
```

luke-mcconnell replied:
Redundancy

mzuhl replied:
Fixed

luke-mcconnell commented on BDF-Chainlit-Chatbot/PDC_tools.py:
```diff
@@ -0,0 +1,582 @@
+#from llama_index.llms.openai import OpenAI
+import json, requests
+from io import StringIO
+from typing import List
```
```suggestion
```

luke-mcconnell replied:
Redundancy

mzuhl replied:
Removed redundancy

luke-mcconnell commented on BDF-Chainlit-Chatbot/PDC_tools.py:
```diff
@@ -0,0 +1,582 @@
+import json, requests
+from io import StringIO
+from typing import List
+import pandas as pd;
```
```suggestion
import pandas as pd
```

mzuhl replied:
Fixed typo

mzuhl committed:
Updates after code review

Updated comments, added and updated environmental variables, fixed typos and redundancies in import, removed logging for now as it was not implemented completely

anandbasu commented on README.md:
```diff
@@ -1,2 +1,22 @@
+
+Clone the repository
+`cd BDF-Chainlit-Chatbot`
+`pip install requirements.txt`
```
Needs a -r in front of requirements.txt

mzuhl replied:
Updated

anandbasu commented on README.md:
```diff
@@ -1,2 +1,22 @@
+`pip install requirements.txt`
+
+If running the app in local environment we suggest to create a virtual environment and run the app from there:
+
```
Add instructions for setting environment variables like PDC_API

anandbasu commented on README.md:
```diff
@@ -1,2 +1,22 @@
+Python version >= 3.10
+
+Before using the app, need to obtain various authentications and set them in .env file
+
```
I think some more details here would be helpful

mzuhl replied:
Elaborated a little on which information might be needed

anandbasu commented on README.md:
```diff
@@ -1,2 +1,22 @@
+
+If running the app in local environment we suggest to create a virtual environment and run the app from there:
+
+
```
Some more instructions may be helpful - I get the following when I run it:
ValueError: You must provide a JWT secret in the environment to use authentication. Run `chainlit create-secret` to generate one.

mzuhl replied:
I have the suggestion on how to generate CHAINLIT_AUTH_SECRET in the example .env file

anandbasu commented on README.md:
```diff
@@ -1,2 +1,22 @@
+
+
+## Running
+
```
Maybe add a comment to set things like ENABLE_PASSWORD_AUTH to false for local testing?

mzuhl replied:
Updated

anandbasu commented on BDF-Chainlit-Chatbot/PDC_tools.py:
```diff
@@ -0,0 +1,582 @@
+    Returns all the matching studies including disease name, primary site and study id.
+    """
+    if None in mdata['disease_type'] and None in mdata['primary_site']:
+        return {'mdata':{}, 'original_message':inp['original_message'], 'data':{}}
```
What is inp?

mzuhl replied:
I am not sure where does it come from, looks like some kind of input variable, maybe forgot to add it in function definition

anandbasu commented on BDF-Chainlit-Chatbot/PDC_tools.py:
```diff
@@ -0,0 +1,582 @@
+
+def get_external_genomic_data(study_id: str, gene_names: List):
+    """ Get data from GDC based on study name"""
+    ensmbl_ids = gene_name_to_ensembl_mapping(gene_names)
```
This function is defined in tools.py - need to import tools

mzuhl replied:
Fixed

mzuhl committed:
Fixed GDC query and updates to README with additional instructions
